api/src/test/java/org/openmrs/CohortTest.java

```diff
@@ -165,4 +168,34 @@ public void setMemberIds_shouldSupportLargeCohorts() {
         double secondsToSet = (endTime - startTime)/1000;
         Assert.assertTrue("Setting cohort of size " + cohortSize + " took " + secondsToSet + " seconds", secondsToSet < 5);
     }
+
+	@Test
+	public void intersect_shouldNotContainDuplicatePatientIDs() throws Exception {
+
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date startDate2 = dateFormat.parse("20017-02-15 00:00:00");
+
+		Cohort cohortA = new Cohort(3);
+		Cohort cohortB = new Cohort (4);
+
+		CohortMembership membershipAOne = new CohortMembership(7, startDate);
+		CohortMembership membershipATwo = new CohortMembership(7, startDate2);
```

Comment:
This seems like an invalid state to me. You should be allowed to be in the same cohort twice, but not with overlapping dates. E.g. you'd need to end membershipAOne before starting membershipATwo.